scrapers_library/data_portals/ckan/scrape_ckan_data_portals.py

```diff
@@ -0,0 +1,284 @@
+"""Retrieves packages from CKAN data portals and parses relevant information then outputs to a CSV file"""
+from itertools import chain
+import json
+import sys
+from typing import Any, Callable, Optional
+
+from from_root import from_root
+import pandas as pd
+from tqdm import tqdm
+
+p = from_root("CONTRIBUTING.md").parent
+sys.path.insert(1, str(p))
+
+from scrapers_library.data_portals.ckan.ckan_scraper_toolkit import (
+    ckan_package_search,
+    ckan_group_package_show,
+    ckan_collection_search,
+    ckan_package_search_from_organization,
+    Package,
+)
+from search_terms import package_search, group_search, organization_search
+
+
+def perform_search(
+    search_func: Callable,
+    search_terms: list[dict[str, Any]],
+    results: list[dict[str, Any]],
+):
+    """Executes a search function with the given search terms.
+
+    :param search_func: The search function to execute.
+    :param search_terms: The list of urls and search terms.
+    :param results: The list of results.
+    :return: Updated list of results.
+    """
+    key = list(search_terms[0].keys())[1]
+    for search in tqdm(search_terms):
```

Comment:
The use of tqdm is an excellent touch, especially since it can take a little while for some of these to load 👌

Reply:
Yes! It used to take ~30 minutes to run before I limited the number of collections being searched for but I figured I'd keep it so that anyone running the program doesn't thin it's frozen